README.md

```diff
@@ -1,28 +1,229 @@
 
 ## Installation
+
 Add this line to your application's Gemfile:
 
 ```ruby
 gem 'scim_rails'
 ```
 
 And then execute:
+
 ```bash
 $ bundle
 ```
 
 Or install it yourself as:
+
 ```bash
 $ gem install scim_rails
 ```
 
+Generate the config file with:
+
+```bash
+$ rails generate scim_rails config
+```
+
+The config file will be located at:
+
+```
+config/initializers/scim_rails_config.rb
+```
+
+Please update the config file with the models and attributes of your app.
+
+Mount the gem in your routes file:
+
+```ruby
+Application.routes.draw do
+  mount ScimRails::Engine => "/"
+end
+```
+
+This will enable the following routes for the Gem to use:
+
+| Request | Route               |
+|:-------:|:-------------------:|
+| get     | 'scim/v2/Users'     |
+| post    | 'scim/v2/Users'     |
+| get     | 'scim/v2/Users/:id' |
+| put     | 'scim/v2/Users/:id' |
+| patch   | 'scim/v2/Users/:id' |
+
+Note: This Gem can be mounted to any path. For example:
+
+```
+https://scim.example.com/scim/v2/Users
+https://www.example.com/scim/v2/Users
+https://example.com/example/scim/v2/Users
+```
+
+## Usage
+
+### List
+
+##### All
+
+Sample request:
+
+```bash
+$ curl -X GET http://username:password@localhost:3000/scim/v2/Users
+```
+
+##### Pagination
+
+This Gem provides two pagination filters; `startIndex` and `count`.
+
+`startIndex` is the positional number you would like to start at. This parameter can accept any integer but anything less than 1 will be interpreted as 1. If you visualize an array with all your user records in the array, `startIndex` is basically what element you would like to start at. If you are familiar with SQL this parameter is directly correlated to the query offset. **The default value for this fitler is 1.**
+
+`count` is the number of records you would like present in the response. **The default value for this filter is 100.**
+
+Sample request:
+
+```bash
+$ curl -X GET http://username:password@localhost:3000/scim/v2/Users?startIndex=38&count=44
+```
+
+Pagination only really works with a determinate order. What that means is, every time you call the database you need to get the results in the exact same order. So the 4th record is _always_ the 4th record and never appears in a different position. If there is no order then records might show up on multiple pages. **The default order is by id** but this can be configured with `scim_users_list_order`.
+
+The pagination filters may be used on their own or in addition to the query filters listed in the next section.
+
+##### Querying
+
+Currently the only filter supported is a single level `eq`. More operators can be added failry easily in future releases. The SCIM RFC documents nested querying which is something we would like to implement in the future.
+
+**Queryable attributes can be mapped in the configuration file.**
+
+Supported filters:
+
+```
+filter=email eq test@example.com
+fitler=userName eq test@example.com
```

Comment:
I didn't think about case sensitivity since Okta specifically will only search against `userName` which in our case is email and `citext` so case doesn't matter.

[RFC 7644 3.4.2.2](https://tools.ietf.org/html/rfc7644#section-3.4.2.2) describes how filtering should work. Upon rereading the `eq` operator is defined as
```
The attribute and operator values must be identical for a match.
```
which I take to mean case-sensitive.

I did notice an issue though that the attribute needs to be case-insensitive. I.e. `userName` can also be `Username` or `UserName` and we are not handling that scenario in this PR.

Do you think I should revise here or open a new issue?

Reply:
That does sound like case-sensitive. If we only need `userName` for OKTA then we can just open an issue and move on.